src/App.tsx

```diff
@@ -1,203 +1,84 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
+import React, { useState, useEffect } from 'react';
+import { Filter } from './components/Filter';
+import { TodoList } from './components/TodoList';
+import { Footer } from './components/Footer';
+import { ErrorMessage } from './components/ErrorMessage';
+import { Todo } from './types/Todo';
+import { getTodos } from './api/todos';
 
-const USER_ID = 0;
+export type FilterType = 'all' | 'active' | 'completed';
+
+const USER_ID = 12004;
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
-  }
+  const [todos, setTodos] = useState<Todo[] | null>(null);
+  const [filteredTodos, setFilteredTodos] = useState<Todo[] | null>(null);
+  const [query, setQuery] = useState<FilterType>('all');
+  const [errorText, setErrorText] = useState<string>('');
+
+  const getTodosList = async () => {
+    try {
+      const tmp = await getTodos(USER_ID);
+
+      setTodos(tmp);
+      setFilteredTodos(tmp);
+    } catch (error) {
+      setErrorText('Unable to load todos');
+    }
+  };
+
+  useEffect(() => {
+    getTodosList();
+  }, []);
+
+  useEffect(() => {
+    setTimeout(() => {
+      setErrorText('');
+    }, 3000);
+  }, [errorText]);
+
+  const filtering = (qu: FilterType) => {
+    setQuery(qu);
+    if (todos) {
+      if (qu === 'active') {
+        setFilteredTodos(todos.filter((el) => !el.completed));
+      } else if (qu === 'completed') {
+        setFilteredTodos(todos.filter((el) => el.completed));
+      } else {
+        setFilteredTodos(todos);
+      }
```

Comment:
Use a switch case here instead of if/else if, also will be better if you create a enum for these magic words and use it everywhere